instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt

```diff
@@ -110,99 +102,69 @@ class RouteRefreshTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.ja
     }
 
     @Test
-    fun expect_route_refresh_to_update_traffic_annotations_and_incidents() {
-        // Request a route.
-        val routes = requestDirectionsRouteSync(coordinates).reversed()
-
-        // Create an observer resource that captures the routes.
-        val initialRouteIdlingResource = RoutesObserverIdlingResource(mapboxNavigation)
-            .register()
-
-        // Set navigation with the route.
-        runOnMainSync {
-            mapboxNavigation.setRoutes(routes)
+    fun expect_route_refresh_to_update_traffic_annotations_and_incidents_for_all_routes() =
+        sdkTest {
+            val routeOptions = generateRouteOptions(coordinates)
+            val requestedRoutes = mapboxNavigation.requestRoutes(routeOptions)
+                .getSuccessfulResultOrThrowException()
+                .routes
+                .reversed()
+
+            mapboxNavigation.setNavigationRoutes(requestedRoutes)
             mapboxNavigation.startTripSession()
-            mockLocationReplayerRule.loopUpdate(
-                mockLocationUpdatesRule.generateLocationUpdate {
-                    latitude = coordinates[0].latitude()
-                    longitude = coordinates[0].longitude()
-                    bearing = 190f
-                },
-                times = 60
-            )
-            mapboxNavigation.registerRouteProgressObserver { routeProgress ->
-                logD(
-                    "progress state: ${routeProgress.currentState}",
-                    LOG_CATEGORY
-                )
-                logD(
-                    "progress duration remaining: ${routeProgress.durationRemaining}",
-                    LOG_CATEGORY
-                )
-            }
-        }
-
-        // Wait for the initial route.
-        val initialRoutes = initialRouteIdlingResource.next()
+            stayOnInitialPosition()
+            val routeUpdates = mapboxNavigation.routesUpdates()
+                .take(2)
+                .map { it.navigationRoutes }
+                .toList()
+            val initialRoutes = routeUpdates[0]
+            val refreshedRoutes = routeUpdates[1]
+
+            mapboxNavigation.routeProgressUpdates()
+                .filter { routeProgress -> isRefreshedRouteDistance(routeProgress) }
+                .first()
+            mapboxNavigation.roadObjectsOnRoute()
+                .filter { upcomingRoadObjects ->
+                    upcomingRoadObjects.size == 2 &&
```

Comment:
why does one need to check the size of the road objects list?

Reply:
You see a test migrated to coroutines from callback. Here's how it looked before the migration 🙂 
<img width="1123" alt="image" src="https://user-images.githubusercontent.com/6190346/175554387-165e3e3e-6709-4c1a-9220-ad64f9099f09.png">

If you changed your opinion, I'm okay to remove that check 🙂

Reply:
ok, we wait for an update with 2 incidents 👍